benches/fibonacci.rs

```diff
@@ -6,135 +6,263 @@ use criterion::{
+                        );
+                        let _ = black_box(result);
+                    },
+                    BatchSize::LargeInput,
+                )
+            },
+        );
     }
 }
 
-fn fibo_prove<M: measurement::Measurement>(
-    prove_params: ProveParams,
-    c: &mut BenchmarkGroup<'_, M>,
-    state: Rc<RefCell<State>>,
-) {
-    let ProveParams {
-        fib_n,
-        reduction_count,
-    } = prove_params;
-
-    let limit = fib_limit(fib_n, reduction_count);
-    let lang_pallas = Lang::<pallas::Scalar, Coproc<pallas::Scalar>>::new();
-    let lang_rc = Arc::new(lang_pallas.clone());
-
-    // use cached public params
-    let instance = Instance::new(
-        reduction_count,
-        lang_rc.clone(),
-        true,
-        Kind::NovaPublicParams,
-    );
-    let pp =
-        public_params::<_, _, MultiFrame<'_, _, _>>(&instance, Utf8Path::new(PUBLIC_PARAMS_PATH))
-            .unwrap();
-
-    c.bench_with_input(
-        BenchmarkId::new(prove_params.name(), fib_n),
-        &prove_params,
-        |b, prove_params| {
-            let store = Store::default();
-
-            let env = empty_sym_env(&store);
-            let ptr = fib::<pasta_curves::Fq>(
-                &store,
-                state.clone(),
-                black_box(prove_params.fib_n as u64),
-            );
-            let prover = NovaProver::new(prove_params.reduction_count, lang_pallas.clone());
-
-            let frames = &prover
-                .get_evaluation_frames(ptr, env, &store, limit, lang_rc.clone())
-                .unwrap();
-
-            b.iter_batched(
-                || (frames, lang_rc.clone()),
-                |(frames, lang_rc)| {
-                    let result = prover.prove(&pp, frames, &store, &lang_rc);
-                    let _ = black_box(result);
-                },
-                BatchSize::LargeInput,
-            )
-        },
-    );
+mod lem {
+    use lurk::lem::{eval::evaluate, multiframe::MultiFrame, pointers::Ptr, store::Store};
+
+    use super::*;
+
+    fn fib<F: LurkField>(store: &Store<F>, state: Rc<RefCell<State>>) -> Ptr<F> {
+        let program = r#"
+(letrec ((next (lambda (a b) (next b (+ a b))))
+           (fib (next 0 1)))
+  (fib))
+"#;
+
+        store.read(state, program).unwrap()
+    }
+
+    pub fn prove<M: measurement::Measurement>(
+        prove_params: ProveParams,
+        c: &mut BenchmarkGroup<'_, M>,
+        state: Rc<RefCell<State>>,
+    ) {
+        let ProveParams {
+            folding_steps,
+            reduction_count,
+            version,
+        } = prove_params;
+
+        assert_eq!(version, Version::LEM);
+        let limit = reduction_count * (folding_steps + 1);
+
+        // Track the approximate it/s. Note that this is not an accurate count because we include the first Nova step,
+        // but that this does get closer to the real value as we run more iterations
+        c.throughput(criterion::Throughput::Elements(limit as u64));
```

Comment:
I know you acknowledge this is not exact, so maybe this is how we want it. However, if the goal is to primarily track folding performance, which we consider to dominate, then maybe we should not use `limit` but rather `reduction-count * folding_steps`.